ddtrace/settings/config.py

```diff
@@ -607,16 +627,72 @@ def __setattr__(self, key, value):
         if key == "_config":
             return super(self.__class__, self).__setattr__(key, value)
         elif key in self._config:
-            self._config[key].set_code(value)
-            self._notify_subscribers([key])
+            self._set_config_items([(key, value, "code")])
             return None
         else:
             return super(self.__class__, self).__setattr__(key, value)
 
+    def _set_config_items(self, items):
+        # type: (List[Tuple[str, Any, _ConfigSource]]) -> None
+        for key, value, origin in items:
+            self._config[key].set_value_source(value, origin)
+
+        from ..internal.telemetry import telemetry_writer
+
+        cs = [{"name": k, "value": self._config[k].value(), "origin": self._get_source(k)} for k, _, _ in items]
+        telemetry_writer.add_event({"configuration": cs}, "app-client-configuration-change")
+        self._notify_subscribers([i[0] for i in items])
+
     def _reset(self):
         # type: () -> None
         self._config = _default_config()
 
     def _get_source(self, item):
         # type: (str) -> str
         return self._config[item].source()
+
+    def _remoteconfigPubSub(self):
+        from ddtrace.internal.remoteconfig._connectors import PublisherSubscriberConnector
+        from ddtrace.internal.remoteconfig._publishers import RemoteConfigPublisher
+        from ddtrace.internal.remoteconfig._pubsub import PubSub
+        from ddtrace.internal.remoteconfig._pubsub import RemoteConfigSubscriber
+
+        class _GlobalConfigPubSub(PubSub):
+            __publisher_class__ = RemoteConfigPublisher
+            __subscriber_class__ = RemoteConfigSubscriber
+            __shared_data__ = PublisherSubscriberConnector()
+
+            def __init__(self, callback):
+                self._publisher = self.__publisher_class__(self.__shared_data__, None)
+                self._subscriber = self.__subscriber_class__(self.__shared_data__, callback, "GlobalConfig")
+
+        return _GlobalConfigPubSub
+
+    def _handle_remoteconfig(self, data, test_tracer=None):
+        # type: (Any, Any) -> None
+
+        # If no data is submitted then the RC config has been deleted. Revert the settings.
+        if not data:
+            for item in self._config.values():
+                item.unset_rc()
+            return
+
+        config = data["config"][0]
+        if "lib_config" not in config:
+            log.warning("unexpected RC payload %r", config)
+            return
+
+        lib_config = config["lib_config"]
+        updated_items = []  # type: List[Tuple[str, Any]]
+
+        if "tracing_sampling_rate" in lib_config:
+            updated_items.append(("_trace_sample_rate", lib_config["tracing_sampling_rate"]))
```

Comment:
Will we need to enumate every configuration that is supported by rcm? Can we avoid this?

Reply:
Like handle each option individually? Yes we will need to since we will potentially need to parse configuration items (like tags)